internal/controller/providerconfig/healthcheck/healthcheck_controller.go

```diff
@@ -199,70 +158,60 @@ func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
+		c.log.Info("Backend client not found during validation bucket cleanup - aborting cleanup", consts.KeyBackendName, req.Name)
 
 		return nil
 	}
 
-	g := new(errgroup.Group)
-	g.Go(func() error {
-		c.log.Info("Deleting health check bucket", consts.KeyBucketName, bucketName, consts.KeyBackendName, req.Name)
-		if err := rgw.DeleteBucket(ctx, backendClient, aws.String(bucketName), true); err != nil {
-			return errors.Wrap(err, errDeleteHealthCheckBucket)
-		}
-
-		return nil
-	})
-
-	g.Go(func() error {
-		c.log.Info("Deleting lifecycle configuration validation bucket", consts.KeyBucketName, v1alpha1.LifecycleConfigValidationBucketName, consts.KeyBackendName, req.Name)
-		if err := rgw.DeleteBucket(ctx, backendClient, aws.String(v1alpha1.LifecycleConfigValidationBucketName), true); err != nil {
-			return errors.Wrap(err, errDeleteLCValidationBucket)
-		}
-
-		return nil
-	})
+	c.log.Info("Deleting lifecycle configuration validation bucket", consts.KeyBucketName, v1alpha1.LifecycleConfigValidationBucketName, consts.KeyBackendName, req.Name)
+	if err := rgw.DeleteBucket(ctx, backendClient, aws.String(v1alpha1.LifecycleConfigValidationBucketName), true); err != nil {
+		return errors.Wrap(err, errDeleteLCValidationBucket)
+	}
 
-	return g.Wait()
+	return nil
 }
 
-// doHealthCheck performs a PutObject and GetObject on the health check bucket on the backend.
-func (c *Controller) doHealthCheck(ctx context.Context, providerConfig *apisv1alpha1.ProviderConfig, bucketName string, o ...func(*s3.Options)) error {
+// doHealthCheck performs a basic http request to the hostbase address.
+func (c *Controller) doHealthCheck(ctx context.Context, providerConfig *apisv1alpha1.ProviderConfig) error {
 	ctx, span := otel.Tracer("").Start(ctx, "Controller.doHealthCheck")
 	defer span.End()
 
-	s3BackendClient := c.backendStore.GetBackendS3Client(providerConfig.Name)
-	if s3BackendClient == nil {
-		return errors.New(errBackendNotStored)
-	}
+	address := utils.ResolveHostBase(providerConfig.Spec.HostBase, providerConfig.Spec.UseHTTPS)
 
-	if putErr := rgw.PutObject(ctx, s3BackendClient, &s3.PutObjectInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(healthCheckFile),
-		Body:   strings.NewReader(time.Now().Format(time.RFC850))}, o...,
-	); putErr != nil {
-		putErr = errors.Wrap(putErr, errPutHealthCheckFile)
-		traces.SetAndRecordError(span, putErr)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, address, http.NoBody)
+	if err != nil {
+		msg := "failed to create request for health check"
+		reqErr := errors.Wrap(err, msg)
+		traces.SetAndRecordError(span, reqErr)
 
-		return putErr
+		return reqErr
 	}
 
-	_, getErr := rgw.GetObject(ctx, s3BackendClient, &s3.GetObjectInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(healthCheckFile),
-	}, o...)
-	if getErr != nil {
-		getErr = errors.Wrap(getErr, errGetHealthCheckFile)
-		traces.SetAndRecordError(span, getErr)
+	resp, err := c.httpClient.Do(req)
```

Comment:
As i see it would use the default timeout of go: `0`

Reply:
:+1:  changed it to use S3 timeout - which is what the original health check would have been using.